lib/src/components/NxCombobox/NxCombobox.tsx

```diff
@@ -87,29 +87,36 @@ function NxComboboxRender<T extends string | number | DataItem<string | number,
         [alertDropdownId]: showAlert
       }),
       TextInput = filterInput ? NxFilterInput : NxTextInput,
-      filterInputProps = filterInput === 'search' ? { searchIcon: true } : null;
+      filterInputProps = filterInput === 'search' ? { searchIcon: true } : null,
+      [pageIsFocused, setPageIsFocused] = useState(true);
+
+  function handlePageFocus() {
+    const onPage = document.hasFocus();
+    onPage ? setPageIsFocused(true) : setPageIsFocused(false);
+  }
 
   // There is a requirement that when there is an error querying the data, if the user navigates away from
   // the component and then comes back to it the search should be retried automatically
   function handleComponentFocus(evt: FocusEvent<HTMLDivElement>) {
-    setInputIsFocused(true);
 
+    setInputIsFocused(true);
     if (loadError) {
       // check if this is focus coming into the component from somewhere else on the page, not just moving between
       // children of this component and not from focus coming into the browser from some other window
-      const comingFromOutsidePage = evt.relatedTarget === null,
+      const comingFromOutsidePage = pageIsFocused === false,
           comingFromChildNode = evt.relatedTarget instanceof Node && evt.currentTarget.contains(evt.relatedTarget);
 
       if (!(comingFromOutsidePage || comingFromChildNode)) {
         doSearch(value);
       }
     }
-
+    handlePageFocus();
```

Comment:
Instead of the extra `handlePageFocus` method, we can replace this with `setPageIsFocused(document.hasFocus());`

Reply:
> Instead of the extra `handlePageFocus` method, we can replace this with `setPageIsFocused(document.hasFocus());`

Thanks, that makes much more sense! [00e3956](https://github.com/sonatype/sonatype-react-shared-components/pull/961/commits/00e3956f5313c10c247493aa3b5ed53aea7f4fbc)